client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -200,8 +210,144 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        while (true) {
+            val state = getState();
+            switch (state.getConfigState()) {
+                case INITIALIZING:
+                    doInit(state);
+                    break;
+                case REINITIALIZING:
+                    doReinit(state);
+                    break;
+                case READY:
+                    long newGen = state.getGeneration() + 1;
```

Comment:
Why do we need to keep a generation counter?

Reply:
We need the generation to make sure that stale API calls are not made to the controller. So for example, if we make a controller call with generation/version 5 then if at the same time someone else was trying to make a call with generation 4 then the call with 4 would be ignored.